contracts/modules/Offers/V1/OffersV1.sol

```diff
@@ -0,0 +1,428 @@
+    //     |                  |                              |
+    //     |                  |        transferFrom()        |
+    //     |                  | ----------------------------->
+    //     |                  |                              |
+    //     |                  |                              |----.
+    //     |                  |                              |    | transfer tokens into escrow
+    //     |                  |                              |<---'
+    //     |                  |                              |
+    //     |                  |----.                         |
+    //     |                  |    | offer count ++          |
+    //     |                  |<---'                         |
+    //     |                  |                              |
+    //     |                  |----.                         |
+    //     |                  |    | create offer            |
+    //     |                  |<---'                         |
+    //     |                  |                              |
+    //     |                  |----.
+    //     |                  |    | offersFor[NFT].append(id)
+    //     |                  |<---'
+    //     |                  |                              |
+    //     |                  |----.                         |
+    //     |                  |    | emit OfferCreated()     |
+    //     |                  |<---'                         |
+    //     |                  |                              |
+    //     |        id        |                              |
+    //     | <-----------------                              |
+    //   Caller           ,---+----.               ,---------+---------.
+    //     ,-.            |OffersV1|               |ERC20TransferHelper|
+    //     `-'            `--------'               `-------------------'
+    //     /|\
+    //      |
+    //     / \
+    /// @notice Creates an offer for an NFT
+    /// @param _tokenContract The address of the desired ERC-721 token
+    /// @param _tokenId The ID of the desired ERC-721 token
+    /// @param _currency The address of the ERC-20 token offering, or address(0) for ETH
+    /// @param _amount The amount offering
+    /// @param _findersFeeBps The bps of the amount (post-royalties) to send to a referrer of the sale
+    /// @return The ID of the created offer
+    function createOffer(
+        address _tokenContract,
+        uint256 _tokenId,
+        address _currency,
+        uint256 _amount,
+        uint16 _findersFeeBps
+    ) external payable nonReentrant returns (uint256) {
+        require(_findersFeeBps <= 10000, "createOffer finders fee bps must be less than or equal to 10000");
+
+        // Validate offer and take custody
+        _handleIncomingTransfer(_amount, _currency);
+
+        // "the sun will devour the earth before it could ever overflow" - @transmissions11
+        unchecked {
```

Comment:
I've run some snapshots with and without unchecked, and ... **without** actually gives out better gas-savings.

Although the increment asm is cheaper with `unchecked`, the opcodes after are actually more expensive.

Can someone confirm?  Tried with both forge and dapp snapshot.

Reply:
```
test_ERC20Integration() (gas: -698 (-0.003%)) 
test_DecreaseETHOfferWithERC20() (gas: -698 (-0.004%)) 
test_IncreaseETHOfferWithERC20() (gas: -698 (-0.004%)) 
test_DecreaseETHOffer() (gas: -698 (-0.004%)) 
test_IncreaseETHOffer() (gas: -698 (-0.004%)) 
testRevert_CannotUpdateOfferWithPreviousAmount() (gas: -698 (-0.004%)) 
test_ETHIntegration() (gas: -698 (-0.004%)) 
testRevert_CannotFillInactiveOffer() (gas: -698 (-0.004%)) 
testRevert_CannotCancelInactiveOffer() (gas: -698 (-0.004%)) 
testRevert_CannotUpdateInactiveOffer() (gas: -698 (-0.004%)) 
test_FillNFTOffer() (gas: -698 (-0.005%)) 
testRevert_AcceptCurrencyMustMatchOffer() (gas: -698 (-0.005%)) 
testRevert_AcceptAmountMustMatchOffer() (gas: -698 (-0.005%)) 
testRevert_OnlyTokenHolderCanFillOffer() (gas: -698 (-0.005%)) 
testRevert_CannotIncreaseOfferWithoutAttachingFunds() (gas: -698 (-0.005%)) 
testRevert_OnlySellerCanCancelOffer() (gas: -698 (-0.005%)) 
testRevert_OnlySellerCanUpdateOffer() (gas: -698 (-0.005%)) 
test_CreateOffer() (gas: -698 (-0.005%)) 
testGas_CreateOffer() (gas: -698 (-0.005%)) 
test_CancelNFTOffer() (gas: -698 (-0.007%)) 
testRevert_UserMustApproveModule() (gas: -2462 (-0.071%)) 
testFail_UserMustApproveTransferHelper() (gas: -3851 (-0.099%)) 
test_ERC20Integration() (gas: -59673 (-0.268%)) 
test_ETHIntegration() (gas: -59813 (-0.381%)) 
```

```
diff --git a/contracts/modules/Offers/V1/OffersV1.sol b/contracts/modules/Offers/V1/OffersV1.sol
index c3323dd..c3c36d9 100644
--- a/contracts/modules/Offers/V1/OffersV1.sol
+++ b/contracts/modules/Offers/V1/OffersV1.sol
@@ -162,9 +162,9 @@ contract OffersV1 is ReentrancyGuard, UniversalExchangeEventV1, IncomingTransfer
         _handleIncomingTransfer(_amount, _currency);
 
         // "the sun will devour the earth before it could ever overflow" - @transmissions11
-        unchecked {
+        // unchecked {
             offerCount++;
-        }
+        // }
 
         offers[_tokenContract][_tokenId][offerCount] = Offer({
             maker: msg.sender,
```

Reply:
yeah this is super weird, do u have an asm diff we can look at?

Reply:
asm and iroptimized
[checked.asm.txt](https://github.com/ourzora/v3/files/7976304/checked.asm.txt)
[checked.iropt.txt](https://github.com/ourzora/v3/files/7976305/checked.iropt.txt)
[unchecked.asm.txt](https://github.com/ourzora/v3/files/7976306/unchecked.asm.txt)
[unchecked.iropt.txt](https://github.com/ourzora/v3/files/7976307/unchecked.iropt.txt)